From 339b5a7e5f17a60df1e3d6d2ffe607786ce34978 Mon Sep 17 00:00:00 2001
From: Chirag Shilwant <c-shilwant@ti.com>
Date: Thu, 13 Feb 2025 14:38:01 +0530
Subject: [PATCH 2/3] lite: Update CMakeLists for building shared object
 library

- When building TFLite with CMake, it builds static library
(i.e. libtensorflow-lite.a) by default which isn't self-contained
since all the transitive dependencies are not included [1]

- As documented in [1], there's an update needed in CMake build steps
inorder to generate a shared object library (i.e. libtensorflow-lite.so)

- This patch adds schema_conversion_utils.cc to TFLite source files
& updates CMakeLists.txt to build TFLite as a shared library. Additionally,
it ensures that the properties are set correctly to manage the shared
object version.

[1]: https://ai.google.dev/edge/litert/build/cmake#step_5_build_litert

Upstream-Status: Inappropriate [enable feature]

Signed-off-by: Chirag Shilwant <c-shilwant@ti.com>
---
 tensorflow/lite/CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tensorflow/lite/CMakeLists.txt b/tensorflow/lite/CMakeLists.txt
index b7bb39869b5..92b05e42446 100644
--- a/tensorflow/lite/CMakeLists.txt
+++ b/tensorflow/lite/CMakeLists.txt
@@ -671,6 +671,7 @@ set(_ALL_TFLITE_SRCS
   ${TFLITE_SOURCE_DIR}/kernels/internal/utils/sparsity_format_converter.cc
   ${TFLITE_SOURCE_DIR}/schema/conversion_metadata_generated.h
   ${TFLITE_SOURCE_DIR}/schema/schema_generated.h
+  ${TFLITE_SOURCE_DIR}/schema/schema_conversion_utils.h
   ${TF_SOURCE_DIR}/compiler/mlir/lite/schema/schema_utils.cc
   ${TF_SOURCE_DIR}/compiler/mlir/lite/schema/schema_generated.h
   ${TF_SOURCE_DIR}/compiler/mlir/lite/schema/conversion_metadata_generated.h
@@ -697,7 +698,7 @@ else()
   list(FILTER _ALL_TFLITE_SRCS EXCLUDE REGEX ".*mmap_allocation\\.cc$")
 endif()

-add_library(tensorflow-lite
+add_library(tensorflow-lite SHARED EXCLUDE_FROM_ALL
   ${_ALL_TFLITE_SRCS}
 )
 set(_ALL_TFLITE_HDRS ${_ALL_TFLITE_SRCS})
@@ -773,6 +774,9 @@ if(TFLITE_ENABLE_INSTALL)
   )
 endif()

+# Manage version for shared object (dynamic) library
+set_target_properties(tensorflow-lite PROPERTIES VERSION "${TFLITE_VERSION_MAJOR}")
+
 # The kernel tests.
 if(TFLITE_KERNEL_TEST)
   enable_testing()
--
2.34.1
